components/supervisor/pkg/ports/ports.go

```diff
@@ -740,12 +744,7 @@ func (pm *Manager) Subscribe() (*Subscription, error) {
 func (pm *Manager) getStatus() []*api.PortsStatus {
 	res := make([]*api.PortsStatus, 0, len(pm.state))
 	for port := range pm.state {
-		portStatus := pm.getPortStatus(port)
-		// Filter out ports that not served and not inside `.gitpod.yml`
-		if _, _, ok := pm.configs.Get(port); !ok && !portStatus.Served {
-			continue
-		}
-		res = append(res, portStatus)
+		res = append(res, pm.getPortStatus(port))
```

Comment:
Filtering maybe is alright just we need to keep public ports always

Reply:
It means, port will be gone if user switches a public not served port to private. 🤔

Reply:
Ok, let’s keep it.

Another approach it is to filter ports which can be only local, ie nodejs debug ports. But I’m not sure how it can achieved on supervisor level. VS Code extensions certainly know it.